### PR TITLE
Clear error type

### PR DESCRIPTION
**Pull request by yuema137**
- Add clear error type like `QOSNotFoundError`. So that users could do the following to bypass some of the validators:
```
try:
    job_submission=JobSubmission()
except QOSNotFoundError:
    pass
```

- Add a unittest for batchq

### CODE REVIEW

File: utilix/batchq.py [hunk 1]
```diff
@@ -120,14 +140,22 @@ class JobSubmission(BaseModel):
...
+        description="Define a list of nodes which should be excluded from submission",
     )
     dependency: Optional[str] = Field(
         None, description="Provide list of job ids to wait for before running this job"
     )
-    verbose: bool = Field(False, description="Print the sbatch command before submitting")
+    verbose: bool = Field(
+        False, description="Print the sbatch command before submitting"
+    )
+    bypass_validation: List[str] = Field(
```
Comment (dachengx) [inline utilix/batchq.py:156]: What is the purpose of `bypass_validation`?

**Comment by dachengx** [comment]
Just to keep a record: [slack thread](https://xenonnt.slack.com/archives/C016DM0JPK9/p1712578084709899)

**Comment by yuema137** [comment]
@dachengx In the latest commits I added the option to bypass validators and also the tests for bypass

File: utilix/batchq.py [hunk 2]
```diff
@@ -56,6 +64,15 @@
     "/dali/lgrandi/grid_proxy/xenon_service_proxy:/project2/lgrandi/grid_proxy/xenon_service_proxy",
 ]
 
+class QOSNotFoundError(Exception):
```
Comment (dachengx) [inline utilix/batchq.py:67]: I think then we do not need these designed exceptions?

Comment (yuema137) [reply utilix/batchq.py:67]: I think it's still good to keep those types for future use

**Review by dachengx — CHANGES_REQUESTED** [review]
I think we should also test `submit_job` like you tested `JobSubmission`.

**Comment by yuema137** [comment]
@dachengx I added a test to check if all of the arguments in `JobSubmission` are handled in `submit_job`. Does it look good now?

**Review by dachengx — APPROVED** [review]

**Pull request merged**